test/es6/default-splitscope.js

```diff
@@ -0,0 +1,483 @@
+            assert.areEqual(f4Obj.next().value().next().value, 10, "Function defined in the param scope captures the formal from param scope even if it is captured in the body scope");
+        }
+    },
+    {
+        name: "Split parameter scope with destructuring",
+        body: function () {
+            function f1( {a:a1, b:b1}, c = function() { return a1 + b1; } ) {
+                assert.areEqual(a1, 10, "Initial value of the first destructuring parameter in the body scope should be the same as the one in param scope");
+                assert.areEqual(b1, 20, "Initial value of the second destructuring parameter in the body scope should be the same as the one in param scope");
+                a1 = 1;
+                b1 = 2;
+                assert.areEqual(a1, 1, "New assignment in the body scope updates the first formal's value in body scope");
+                assert.areEqual(b1, 2, "New assignment in the body scope updates the second formal's value in body scope");
+                assert.areEqual(c(), 30, "The param scope method should return the sum of the destructured formals from the param scope");
+                return c;
+            }
+            assert.areEqual(f1({ a : 10, b : 20 })(), 30, "Returned method should return the sum of the destructured formals from the param scope");
+
+            function f2({x:x = 10, y:y = function () { return x; }}) {
+                assert.areEqual(x, 10, "Initial value of the first destructuring parameter in the body scope should be the same as the one in param scope");
+                x = 20;
+                assert.areEqual(x, 20, "Assignment in the body updates the formal's value");
+                return y;
+            }
+            assert.areEqual(f2({ })(), 10, "Returned method should return the value of the destructured formal from the param scope");
+            
+            function f3({y:y = function () { return x; }, x:x = 10}) {
+                assert.areEqual(x, 10, "Initial value of the first destructuring parameter in the body scope should be the same as the one in param scope");
+                x = 20;
+                assert.areEqual(x, 20, "Assignment in the body updates the formal's value");
+                return y;
+            }
+            assert.areEqual(f3({ })(), 10, "Returned method should return the value of the destructured formal from the param scope even if declared after");
+            
+            (({x:x = 10, y:y = function () { return x; }}) => {
+                assert.areEqual(x, 10, "Initial value of the first destructuring parameter in the body scope should be the same as the one in param scope");
+                x = 20;
+                assert.areEqual(y(), 10, "Assignment in the body does not affect the formal captured from the param scope");
+            })({});
+        }
+    },
+    {
+      name: "Nested split scopes",
```

Comment:
did you add the case where you have object literal as default which has the function ?
f(a = 10, b = { foo : function () { return a; } } )